src/pydartdiags/interactive_plots/interactive_plots.py

```diff
@@ -0,0 +1,259 @@
+from dash import Dash, dcc, html, Input, Output, callback
+import pandas as pd
+import plotly.express as px
+import json
+
+dash_styles = {
+    "pre": {"border": "thin lightgrey solid", "overflowX": "scroll", "color": "black"},
+    "txt": {"color": "black"},
+}
+
+
+def geo_plot(obs_seq):
+
+    # Check if 'DART_quality_control' is a column in the dataframe
+    if "DART_quality_control" not in obs_seq.df.columns:
+        print(
+            "'DART_quality_control' is not used in this obs sequence"
+            "file. Color mapping will not be applied."
+        )
+        fig = px.scatter_geo(
+            obs_seq.df,
+            lat="latitude",
+            lon="longitude",
+            hover_data={"type": True, "observation": True, "obs_num": True},
+            width=1600,
+            height=1300,
+        )
+    else:
+        fig = px.scatter_geo(
+            obs_seq.df,
+            lat="latitude",
+            lon="longitude",
+            color="DART_quality_control",
+            hover_data={
+                "DART_quality_control": True,
+                "type": True,
+                "observation": True,
+                "obs_num": True,
+            },
+            width=1400,
+            height=850,
+        )
+        fig.update_layout(coloraxis_colorbar=dict(len=0.8))
+
+    fig.update_layout(clickmode="event+select")
+    # If the obs_seq has more than 10,000 observations, do not show hover info
+    # to reduce strain on the Dash app
+    if len(obs_seq.df) > 10000:
+        fig.update_traces(hoverinfo="skip", hovertemplate=None)
+
+    return fig
+
+
+# Creates Dash app to create a new obs_seq file including only
+# the selected obs
+def include_only_selected_obs_dash_app(fig, obs_seq_selected):
```

Comment:
Can users pass their own figure to this?

Reply:
No the fig is defined in the src code function geo_plot (interactive_plots.py), which creates the geographic plot for the obs sequence that is passed in at [https://github.com/NCAR/pyDARTdiags/blob/31fa261206cbb46d3f4c2fdcf46db8ba101fd6ad/examples/04_interacting/select_to_remove_obs.py#L47](https://github.com/NCAR/pyDARTdiags/blob/31fa261206cbb46d3f4c2fdcf46db8ba101fd6ad/examples/04_interacting/select_to_remove_obs.py#L47)

This bit of code could be moved from the src code to the examples if we want our users to be able to pass their own figures in, but I think that would only be useful if we wanted them to be able to change the hover info and such.

Reply:
@mjs2369 no need to respond to the comments, I closed the pull request